pluma/dialogs/pluma-preferences-dialog.c

```diff
@@ -199,6 +212,88 @@ auto_indent_checkbutton_toggled (GtkToggleButton        *button,
 	pluma_prefs_manager_set_auto_indent (gtk_toggle_button_get_active (button));
 }
 
+static void
+draw_spaces_checkbutton_toggled (GtkToggleButton        *button,
+                                 PlumaPreferencesDialog *dlg)
+{
+	int v;
+        pluma_debug (DEBUG_PREFS);
+
+        g_return_if_fail (button == GTK_TOGGLE_BUTTON (dlg->priv->draw_spaces_checkbutton));
+
+        if (gtk_toggle_button_get_active (button))
+                v = DRAW_ALL;
+        else
+                v = DRAW_NONE;
+
+        pluma_prefs_manager_set_draw_spaces (v);
+#ifdef GTK_SOURCE_VERSION_3_24
+        gtk_widget_set_sensitive (GTK_WIDGET (dlg->priv->draw_trailing_spaces_checkbutton), v > DRAW_NONE);
+        gtk_toggle_button_set_inconsistent (GTK_TOGGLE_BUTTON (dlg->priv->draw_trailing_spaces_checkbutton), v == DRAW_NONE);
+#endif
+
+}
+
+static void
+draw_trailing_spaces_checkbutton_toggled (GtkToggleButton        *button,
```

Comment:
One minor thing I noticed with this function. If you follow these steps:
1. check `draw_spaces_checkbutton`
2. check `draw_trailing_spaces_checkbutton`
3. uncheck `draw_spaces_checkbutton`
4. check `draw_spaces_checkbutton`

You'll end up with `draw_trailing_spaces_checkbutton` checked, but non-trailing spaces are drawn.

Two possible solutions are:
* When you uncheck `draw_spaces_checkbutton`, also uncheck `draw_trailing_spaces_checkbutton` before disabling it.
* When you check `draw_spaces_checkbutton`, verify whether `draw_trailing_spaces_checkbutton` is checked and set the option accordingly

All this also applies to the `tabs` checkboxes.

Reply:
Submitted a commit (b21161b2387d0859061aa8d4ada3daa5b179d74a) to address this